### DISCUSSION TITLE
Compile the Neptune 3 firmware

### DISCUSSION THREAD

**Discussion by papybg** [NARUTOfzr/Neptune_3#67] [Q&A]
I am using your latest beta version on my printer. I wanted to make a small temperature correction - instead of 275 degrees, write 300 degrees. I used the source code you provided which I modified and recompiled. Everything is fine, but the hotend cooling fan does not want to turn on. Where is the problem and how can I fix it.
I am new to this field.

**Comment by NARUTOfzr** [comment]
The fan at the hot end starts only when the temperature at the hot end exceeds 50 ° C.

Please send me the firmware you compiled so that I can confirm whether this is a firmware problem.




---Original---
From: ***@***.***&gt;
Date: Thu, Apr 6, 2023 03:50 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [NARUTOfzr/Neptune_3] Compile the Neptune 3 firmware (Discussion #67)




 
I am using your latest beta version on my printer. I wanted to make a small temperature correction - instead of 275 degrees, write 300 degrees. I used the source code you provided which I modified and recompiled. Everything is fine, but the hotend cooling fan does not want to turn on. Where is the problem and how can I fix it.
 I am new to this field.

**Comment by papybg** [comment]
Sorry for the question, but how exactly do I send you the file?

**Reply by NARUTOfzr** [reply]
Here you can upload the compressed.zip file. You need to compress the compiled "ZNP_ROBIN_NANO.bin" file and upload it.

**Comment by NARUTOfzr** [comment]
![image](https://user-images.githubusercontent.com/64059773/230283724-7e32144e-113b-46a1-9c4b-1e0e9b341bfd.png)
This is the setting that controls the hot end fan. At present, I have not modified these Settings in the source code I uploaded, and the cooling fan at the hot end can only be started after the temperature exceeds 50℃.

Does your machine not turn the cooling fan after preheating the temperature of the hot end? Can you feed back the fan corresponding to the picture?

**Comment by papybg** [comment]
With the original firmware from you, the fan works, but when I compile my firmware, it refuses to start even when preheated and when printing above 50 degrees

**Comment by papybg** [comment]
[ZNP_ROBIN_NANO.zip](https://github.com/NARUTOfzr/Neptune_3/files/11167242/ZNP_ROBIN_NANO.zip)

**Comment by papybg** [comment]
I found that in pins_MKS_E3_V2.h PA7 is disabled, I removed the dashes but it still won't start.

**Comment by papybg** [comment]
To begin with, could you please compile me a bin file with the following changes:
1. hotend temperature 300 degrees
2. bed temperature 120 degrees
3. during a pause, the bed should go forward, and not backwards as it is now, in order to be able to work on the seal.

**Comment by NARUTOfzr** [comment]
[ZNP_ROBIN_NANO-20230406.zip](https://github.com/NARUTOfzr/Neptune_3/files/11167541/ZNP_ROBIN_NANO-20230406.zip)

**Comment by NARUTOfzr** [comment]
I hope this helps you.
It should be noted that the original hardware configuration of the machine does not support high temperature (hot end does not support &gt; 260℃)

**Comment by papybg** [comment]
Thanks, yes it is, but I've put in a metal hotend that supposedly works up to 400 degrees, and the thermistor I think works up to 300 degrees. So I don't see any problem.

**Comment by papybg** [comment]
Many thanks for the quick replies and the file. Everything is OK.
I have a second question, is there an option to put a second extruder. Can you recommend something and possibly give me installation guidance and compile firmware?

**Comment by NARUTOfzr** [comment]
I haven't tried setting up a second extruder at the moment. This may take some time to try. I recommend you to purchase authentic TMC2209/2225 stepper motor drive made by makerbase. The pins are color-differentiated to facilitate installation.




---Original---
From: ***@***.***&gt;
Date: Thu, Apr 6, 2023 17:01 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Compile the Neptune 3 firmware (Discussion#67)




 
Many thanks for the quick replies and the file. Everything is OK.
 I have a second question, is there an option to put a second extruder. Can you recommend something and possibly give me installation guidance and compile firmware?

**Comment by NARUTOfzr** [comment]
![企业微信截图_16808294807783](https://user-images.githubusercontent.com/64059773/230519948-ea640f42-9faf-4ff6-a664-d7e48cb48a8b.png)
![企业微信截图_16808293362548](https://user-images.githubusercontent.com/64059773/230519960-2e83a996-00c6-434e-bb09-ede9d6ff090a.png)

Only the Settings in the figure are needed to control extruder 1/ extruder 2 using T0/T1.
[ZNP_F401_Marlin-Beta.zip](https://github.com/NARUTOfzr/Neptune_3/files/11175335/ZNP_F401_Marlin-Beta.zip)
